### PR TITLE
BTAT-7749 - Upgrade to Play 26

### PR DESCRIPTION
**Pull request by kylemchownhmrc**

### CODE REVIEW

File: test/testUtils/TestSupport.scala [hunk 1]
```diff
@@ -28,4 +31,7 @@ trait TestSupport extends UnitSpec with GuiceOneServerPerSuite with Materializer
 
   implicit val headerCarrier: HeaderCarrier = HeaderCarrier()
 
+  lazy val cc: ControllerComponents = stubControllerComponents()
+
+  def fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("", "")
 }
```
Comment (Tasmine10ten) [inline test/testUtils/TestSupport.scala:37]: There is a duplicate of 
def fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("", "")

This is in SpecBase & does not have any usage elsewhere - can be removed

File: app/repositories/DataRepository.scala [hunk 2]
```diff
@@ -19,13 +19,18 @@ package repositories
...
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class DataRepository @Inject()() extends MongoDbConnection {
+class DataRepository @Inject()(reactiveMongoComponent: ReactiveMongoComponent) {
+
+  lazy val mongoConnector: MongoConnector = reactiveMongoComponent.mongoConnector
+  implicit lazy val db: () => DefaultDB = mongoConnector.db
 
   private lazy val repository = new DataRepositoryBase()
```
[the remark this answers is not in the document]
Comment (TheDeadlyPianist) [reply app/repositories/DataRepository.scala:36]: It is suspected that this is a warning to a package structure shuffle.
The file is in the correct package structure, but the pack in the file itself is the old one. Therefore is it safe to conclude that the deprecation message is a warning to this changing at some point in the future.

File: app/repositories/SchemaRepository.scala [hunk 3]
```diff
@@ -18,13 +18,18 @@ package repositories
...
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class SchemaRepository @Inject()() extends MongoDbConnection {
+class SchemaRepository @Inject()(reactiveMongoComponent: ReactiveMongoComponent) {
+
+  lazy val mongoConnector: MongoConnector = reactiveMongoComponent.mongoConnector
+  implicit lazy val db: () => DefaultDB = mongoConnector.db
 
   private lazy val repository = new SchemaRepositoryBase()
```
[the remark this answers is not in the document]
Comment (TheDeadlyPianist) [reply app/repositories/SchemaRepository.scala:35]: See above

File: test/testUtils/TestSupport.scala (hunk 1 above)
Comment (TheDeadlyPianist) [reply test/testUtils/TestSupport.scala:37]: SpecBase isn't used anywhere. Deleting it.

**Pull request merged**